packages/modules/src/modules/epics/ContributionsEpic.tsx

```diff
@@ -260,21 +280,35 @@ const ContributionsEpic: React.FC<EpicProps> = ({
     hasConsentForArticleCount,
     stage,
 }: EpicProps) => {
-    const countryGroupId = countryCodeToCountryGroupId(countryCode || 'GBPCountries');
-    const defaultFrequency: ContributionFrequency = variant.defaultChoiceCardFrequency || 'MONTHLY';
-    const [choiceCardSelection, setChoiceCardSelection] = useState<ChoiceCardSelection | undefined>(
-        variant.choiceCardAmounts && {
-            frequency: defaultFrequency,
-            amount:
-                variant.choiceCardAmounts[countryGroupId]['control'][defaultFrequency][
-                    'amounts'
-                ][1],
-        },
-    );
+    const { image, tickerSettings, showChoiceCards, choiceCardAmounts } = variant;
 
-    const { hasOptedOut, onArticleCountOptIn, onArticleCountOptOut } = useArticleCountOptOut();
+    const currencySymbol = getLocalCurrencySymbol(countryCode);
 
-    const { image, tickerSettings, showChoiceCards, choiceCardAmounts } = variant;
+    const [choiceCardSelection, setChoiceCardSelection] = useState<
+        ChoiceCardSelection | undefined
+    >();
+
+    const [amountsVariant, setAmountsVariant] = useState<SelectedAmountsVariant | undefined>();
+
+    useEffect(() => {
```

Comment:
I wonder if the `useEffect` is necessary here? Do we expect `showChoiceCards` or `choiceCardAmounts` to ever change over the lifecycle of this component?

Reply:
Sadly necessary. I originally coded this functionality direct but it led to infinite react render loops. Sticking the functionality into a `useEffect` hook solves that issue.

Reply:
I think the useEffect is needed because we're initialising the `choiceCardSelection` state here.
An alternative to a useEffect would be to do this initialisation in the `useState`, e.g.:

```
const [choiceCardSelection, setChoiceCardSelection] = useState<ChoiceCardSelection | undefined>(
    () => {
        if (showChoiceCards && choiceCardAmounts) {
            const defaultFrequency: ContributionFrequency =
                variant.defaultChoiceCardFrequency || 'MONTHLY';
            const localAmounts = choiceCardAmounts.amounts[defaultFrequency];
            const defaultAmount = localAmounts.defaultAmount || localAmounts.amounts[1] || 1;

            return {
                frequency: defaultFrequency,
                amount: defaultAmount,
            };
        }
        return undefined;
    },
);
```